lib/bolt/module_installer.rb

```diff
@@ -28,7 +28,7 @@ def add(name, modules, puppetfile_path, moduledir, config_path)
       end
 
       # If the Puppetfile exists, make sure it's managed by Bolt.
-      if puppetfile_path.exist?
+      if puppetfile_path.exist? && !force
```

Comment:
I don't think we want to forcibly install here. When adding a single module, the installer resolves dependencies in one of two ways:

- Resolves the Puppetfile's modules with the new module
- Resolves the module declarations with the new module

If there is a mismatch between the module declarations and the Puppetfile, then we can't be sure what the user actually intends by trying to add a module. Are they trying to add a module to a manually edited Puppetfile? Then they should manually edit the Puppetfile instead of using the `module add` command. Are they trying to add a module to a Bolt-managed Puppetfile? Then they should be told to re-install their modules with `module install --force`.

Reply:
I think another option may be to skip the initial attempt to resolve with the Puppetfile's modules when forcibly adding a module, but I still think if you are using Bolt to manage your modules and want to use the `module add` command, then your Puppetfile should be managed by Bolt.

Reply:
I only initially added this because I saw that `force` is listed as an option for `module add` https://github.com/puppetlabs/bolt/blob/main/lib/bolt/bolt_option_parser.rb#L70 but to me it makes sense to have the force option available here to both add the module to the modules list and transition the Puppetfile to a managed Puppetfile, just like with module install. We would know the user is trying to add a module to a Bolt-managed Puppetfile when they ran with `--force`, as they simply wouldn't run this with that flag if they wanted to manually add a module to their Puppetfile.

Reply:
Ah, that's listed as an option because I originally added support for `--force` but then removed it for the above reasons.

If we move forward with `module add --force` then this would need to be updated to skip resolving with the Puppetfile's modules: https://github.com/puppetlabs/bolt/blob/main/lib/bolt/module_installer.rb#L71-L84

Reply:
M'k - we can discuss at standup tomorrow